config/console.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python
+
+import click
+
+import utilities_common.cli as clicommon
+
+#
+# 'console' group ('config console ...')
+#
+@click.group('console')
+def console():
+    """Console-related configuration tasks"""
+    pass
+
+#
+# 'console add' group ('config console add ...')
+#
+@console.command('add')
+@clicommon.pass_db
+@click.argument('linenum', metavar='<line_number>', required=True, type=click.IntRange(0, 65535))
+@click.option('--baud', '-b', metavar='<baud>', required=True, type=click.INT)
+@click.option('--flowcontrol', '-f', metavar='<flow_control>', required=False, is_flag=True)
+@click.option('--devicename', '-d', metavar='<device_name>', required=False)
+def add_console_setting(db, linenum, baud, flowcontrol, devicename):
+    """Add Console-realted configuration tasks"""
+    config_db = db.cfgdb
+
+    table = "CONSOLE_PORT"
+    dataKey1 = 'baud_rate'
+    dataKey2 = 'flow_control'
+    dataKey3 = 'remote_device'
+
+    ctx = click.get_current_context()
+    data = config_db.get_entry(table, linenum)
+    if data:
+        ctx.fail("Trying to add console port setting, which is already exists.")
+    else:
+        console_entry = { dataKey1: baud }
+        console_entry[dataKey2] = "1" if flowcontrol else "0"
+
+        if devicename:
+            if isExistingSameDevice(config_db, devicename, table):
```

Comment:
Could it be that user are updating an existing configuration and give device name again?

Reply:
We could, but not thru. this command, see HLD: https://github.com/Azure/SONiC/blob/master/doc/console/SONiC-Console-Switch-High-Level-Design.md#33122-update-remote-device-name-for-a-console-port